model/jpa/src/main/java/org/keycloak/models/jpa/UserAdapter.java

```diff
@@ -147,8 +152,9 @@ public void setAttribute(String name, List<String> values) {
         // Remove all existing
         removeAttribute(name);
 
-        // Put all new
-        for (String value : values) {
+        List<String> nonNullValues = values.stream().filter(Objects::nonNull).collect(Collectors.toList());
```

Comment:
No need to realize this list into memory via `.collect`. you can either obtain a stream `iterator()` or use `Stream.forEach()`.